crates/ruma-identifiers/src/user_id.rs

```diff
@@ -100,7 +103,7 @@ impl UserId {
     /// A historical user ID is one that doesn't conform to the latest specification of the user ID
     /// grammar but is still accepted because it was previously allowed.
     pub fn is_historical(&self) -> bool {
-        !localpart_is_fully_conforming(self.localpart()).unwrap()
+        !localpart_is_fully_conforming(self.localpart()).unwrap_or(false)
```

Comment:
This should not be needed anymore if invalid user ids can't be constructed, but it feels safer. Let me know if you would like to maintain it or if I should revert the change.

Reply:
I would like for this change to be reverted; if there is a library bug panicking is the right thing to do.